package indoor_context.proto;

import "vector.proto";

message MatF {
	required int32 rows = 1;
	required int32 cols = 2;
	repeated float entries = 3;
}

message MatI {
	required int32 rows = 1;
	required int32 cols = 2;
	repeated int32 entries = 3;
}

message PayoffFeature {
	required MatF left = 5;
	optional MatF right = 6;  // if omitted then right==left implicitly
	optional string description = 10;  // human-readable description of this feature set
}

message FrameWithFeatures {
	required string sequence = 1;  // name of the sequence
	required int32 frame_id = 2;

	required MatI orients = 10;  // column vector
	required MatI path = 11;  // column vector

	required MatF path_features = 20;  // one row per feature
	required int32 num_walls = 21;
	required int32 num_occlusions = 22;

	repeated PayoffFeature features = 30;  // might have zero elements
}

message PayoffFeatureSet {
	repeated FrameWithFeatures frames = 1;
}
